client_server_test.go

```diff
@@ -42,6 +42,8 @@ var cstDialer = Dialer{
 
 type cstHandler struct{ *testing.T }
 
+type cstContextTakeoverHandler struct{ *testing.T }
```

Comment:
Use `cstHandler` and `cstServer` for all tests.

- Add field to `cstHandler` to specify context takeover options.  
- Add arguments to `newServer` to pass options through to the handler.
- Change `cstHandler.ServeHTTP` to send and receive two messages. 
- Modify `sendRecv` to send and receive two messages.  Use loop over messages instead of duplicating gobs of code.

      for _, message := range []string{"Hello World", "Can you read message?"} {
            ...
      }